src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/CachedDirectoryStoreProvider.cs

```diff
@@ -0,0 +1,68 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using Microsoft.Win32.SafeHandles;
+
+namespace Internal.Cryptography.Pal
+{
+    internal sealed class CachedDirectoryStoreProvider
+    {
+        private static readonly TimeSpan s_lastWriteRecheckInterval = TimeSpan.FromSeconds(1);
+        private static readonly TimeSpan s_assumeInvalidInterval = TimeSpan.FromSeconds(30);
```

Comment:
30 seconds instead of the 5 minutes you mentioned yesterday?  I don't have a strong opinion either way, just curious as to how you arrived at the intervals.

Reply:
5 minutes is the reload for the machine trust (down to 5 minutes from "never").
30 seconds is the reload for user data (up from "always").

That's where the "after 5 minutes we have 2 (system) + 3 (My, CA, Root) * 10 = 32 finalizations enqueued" came from.